app/src/main/java/org/wikipedia/page/SidePanelHandler.kt

```diff
@@ -11,41 +12,61 @@ import android.view.ViewGroup
 import android.webkit.ValueCallback
 import android.widget.*
 import android.widget.AdapterView.OnItemClickListener
+import androidx.core.view.isVisible
 import androidx.core.view.updateLayoutParams
 import androidx.drawerlayout.widget.DrawerLayout
 import androidx.drawerlayout.widget.FixedDrawerLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.textview.MaterialTextView
 import org.json.JSONException
 import org.json.JSONObject
+import org.wikipedia.Constants
 import org.wikipedia.R
 import org.wikipedia.WikipediaApp
 import org.wikipedia.analytics.ToCInteractionFunnel
 import org.wikipedia.bridge.CommunicationBridge
 import org.wikipedia.bridge.JavaScriptActionHandler
+import org.wikipedia.databinding.ItemTalkTopicBinding
 import org.wikipedia.dataclient.WikiSite
-import org.wikipedia.util.DimenUtil
-import org.wikipedia.util.L10nUtil
-import org.wikipedia.util.ResourceUtil
-import org.wikipedia.util.StringUtil
-import org.wikipedia.views.ObservableWebView
+import org.wikipedia.dataclient.mwapi.MwQueryPage
+import org.wikipedia.dataclient.okhttp.HttpStatusException
+import org.wikipedia.dataclient.page.TalkPage
+import org.wikipedia.diff.ArticleEditDetailsActivity
+import org.wikipedia.richtext.RichTextUtil
+import org.wikipedia.settings.Prefs
+import org.wikipedia.talk.TalkTopicHolder
+import org.wikipedia.talk.TalkTopicsActivity
+import org.wikipedia.talk.TalkTopicsProvider
+import org.wikipedia.util.*
+import org.wikipedia.views.*
 import org.wikipedia.views.ObservableWebView.OnContentHeightChangedListener
-import org.wikipedia.views.PageScrollerView
-import org.wikipedia.views.SwipeableListView
 import org.wikipedia.views.SwipeableListView.OnSwipeOutListener
-import java.util.*
 
-class ToCHandler internal constructor(private val fragment: PageFragment,
-                                      private val drawerLayout: FixedDrawerLayout,
-                                      private val scrollerView: PageScrollerView,
-                                      private val bridge: CommunicationBridge) :
+class SidePanelHandler internal constructor(private val fragment: PageFragment,
+                                            private val drawerLayout: FixedDrawerLayout,
+                                            private val scrollerView: PageScrollerView,
+                                            private val bridge: CommunicationBridge) :
         ObservableWebView.OnClickListener, ObservableWebView.OnScrollChangeListener, OnContentHeightChangedListener {
 
+    private val containerView = drawerLayout.findViewById<FrameLayout>(R.id.side_panel_container)
```

Comment:
Instead of getting all these Views through `findViewById()`, can we just pass the `binding` object from PageActivity into here?

Reply:
Done!